src/ui/src/builder/useComponentFieldViewModel.ts

```diff
@@ -0,0 +1,45 @@
+import { computed, inject, Ref, DeepReadonly } from "vue";
+import injectionKeys from "@/injectionKeys";
+import { useComponentActions } from "./useComponentActions";
+
+type Params = {
+	componentId: DeepReadonly<Ref<string>>;
+	fieldKey: DeepReadonly<Ref<string>>;
+	defaultValue?: DeepReadonly<Ref<string>>;
+};
+
+export function useComponentFieldViewModel(params: Params) {
+	const wf = inject(injectionKeys.core);
+	const ssbm = inject(injectionKeys.builderManager);
```

Comment:
It's always better to "inject" dependencies; I'll make your composable more testable and more reusable.

```suggestion
export function useComponentFieldViewModel(
	params: Params,
	wf = inject(injectionKeys.core),
	ssbm = inject(injectionKeys.builderManager),
) {
```

Reply:
> make your composable more testable and more reusable

good pint, I fully agree with that!
but I've not seen that way of setting inject() as default parameters and I could not find that example in Vue documentation or their gihtub. 

If needed for testing, I can still mock the injects by passing them explicitly:

```ts
type Dependencies = {
	wf?: Core;
	ssbm?: BuilderManager;
};

export function useComponentFieldViewModel(
	params: Params,
	dependencies?: Dependencies,
) {
	const wf = dependencies?.wf ?? inject(injectionKeys.core);
	const ssbm = dependencies?.ssbm ?? inject(injectionKeys.builderManager);
```

Reply:
I think the default value to `inject` is supported by Vue.js.

But thinking about it again, I think the standard way in the application is like this

https://github.com/writer/writer-framework/blob/21bb1b15c665e84fcd0689e020441d65f123b071/src/ui/src/core/useNotesManager.ts#L34-L37
https://github.com/writer/writer-framework/blob/21bb1b15c665e84fcd0689e020441d65f123b071/src/ui/src/composables/useBlueprintRun.ts#L103-L107

So I think it should be

```ts
export function useComponentFieldViewModel(
 	wf: ReturnType<typeof generateCore>, 
 	wfbm: ReturnType<typeof generateBuilderManager>, 
	params: Params,
) {
```